### ISSUE TITLE
JsonObject decodes first object in a list of objects (in invalid json string) instead of throwing an error

### ISSUE THREAD

**Issue by mbaeuerle** [eclipse-vertx/vert.x#3260]
### Version

Which version(s) did you encounter this bug ?
3.8.5

### Context

When creating a new JsonObject by passing in a json string, which contains more than one object separated by comma then JsonObject parses the first one and drops the rest.
Expected behaviour: JsonObject throws a DecodeException as the input json is invalid (missing surrounding parenthesis).

### Background

This issue arises also in Vert.x web when retrieving the body of a request as json via: `RoutingContext.getBodyAsJson()` because it uses JsonObject for parsing under the hood. 

### Do you have a reproducer?
```java
final String s = "{\"foo\":1},{\"bar\":2}";
JsonObject obj = new JsonObject(s);
System.out.println(obj);
// Output: {"foo":1}
```

**Comment by sophokles73** [comment]
@mbaeuerle is this still an issue in 3.8.4?

**Comment by mbaeuerle** [comment]
Yes, I have just checked, the issue is also present in 3.8.4 and the newly released 3.8.5.
I will update the version above.
